main.py

```diff
@@ -319,82 +354,54 @@ def main():
     parser.add_argument('--filename', help="Path to file where bootimage data can be recorded; will allow for faster execution if script is run multiple times", default="deleted_images.json")
     args = parser.parse_args()
 
+    global DRY_RUN
+    DRY_RUN = False
+    image_list = {}
+    deleted_images_json = {}
+
+    if args.dry_run:
+        DRY_RUN = True
     if args.debug:
         logging.basicConfig(level=logging.DEBUG)
 
     if args.filename:
         deleted_images_filename = args.filename
 
-    ### testing functions
-    # get aliyun images in the installer
-    bootimages = parse_openshift_installer(args.release)
-
-    # get builds with aliyun uploads from a builds.json
-    aliyun_releases = parse_release(args.release)
-
     # preload images that should be deleted
-    deleted_images_json = {}
     if os.path.exists(deleted_images_filename):
-        logging.debug("Found file")
+        logging.debug(f"Found file: {deleted_images_filename}")
         with open(deleted_images_filename, 'r') as f:
             deleted_images_json = json.load(f)
 
+    # get aliyun images in the installer
+    bootimages = parse_openshift_installer(args.release)
+    bootimages = get_images_not_tagged(bootimages)
+    # get builds with aliyun uploads from a builds.json
+    aliyun_releases = parse_release(args.release, deleted_images_json)
+    aliyun_releases = get_images_not_tagged(aliyun_releases)
+
     # find the builds from builds.json that are not in bootimages
     for buildid in aliyun_releases.keys():
         if buildid in bootimages:
             print(f"Build ID {buildid} in bootimage metadata; tagging with bootimage=true")
             for region in aliyun_releases[buildid]:
+                image_id =region['image_id']
+                region = region['region_id']
                 # TODO: uncomment this when we want to go live
                 #tag_image(region, image_id, tag_key="bootimage", tag_value="true")
                 pass
+        elif buildid in builds:
```

Comment:
I believe this should be 
```suggestion
        elif buildid in deleted_images_json:
```